### ISSUE TITLE
Problem with data_dir

### ISSUE THREAD

**Issue by Berranzan** [getnamo/tensorflow-ue4-examples#10]
Examples that have line:
self.data_dir = self.scripts_path + '/dataset/mnist'

have to be changed to:
self.data_dir = '/tmp/tensorflow/mnist/input_data'

in order to execute. Not sure if some functionality is affected but examples seem to be working as intended with the change.

**Comment by getnamo** [comment]
To my knowledge ```mnist = input_data.read_data_sets(self.data_dir, one_hot=True)``` will pull mnist data to the specified directory. It may be that you are placing your examples folder in a place where you do not have write access?

Purpose of the above line is to store the downloaded data inside the _Content/Scripts/dataset/mnist_ path so all data is embedded to your project, but you can change it to whatever you want and it should still work as long as the program has write access to that directory.

**Comment by Berranzan** [comment]
This is more of a heads up for people who have problems running the examples, it is excellent work BTW.
Directory is as specified in plugin .../Scripts/dataset/mnist and I see other examples write in the directory, for example mnistSaveLoad.py creates .../Scripts/model/ no problem. 
mnistKerasCNN.py also uses this with no problem:
                model_file_name = "mnistKerasCNN"
		model_directory = ue.get_content_dir() + "/Scripts/"
		model_sess_path =  model_directory + model_file_name + ".tfsess"
		model_json_path = model_directory + model_file_name + ".json"
Still working on additional requirements:
Install path cannot be too long.
Examples won't work at first run but folder
C:\tensorflow2\Plugins\UnrealEnginePython\Binaries\Win64\Lib
gets populated with subfolder site-packages and you can rerun examples.
I have made a Youtube video showing how to get the examples running:
https://www.youtube.com/watch?v=ZciLnYV4jIo&feature=youtu.be

**Comment by getnamo** [comment]
Interesting observations, this should definitely help with troubleshooting for other devs. Good video :)

You are correct that filepath being too long will cause a problem and I have noticed this before, it also doesn't appear that enabling long filepaths fixes this on windows either sadly. That said, for me it is sufficient to place the examples directly on the desktop or in a custom folder without write restrictions.

A couple of things: 
At the moment there is a bug (tracked as issue https://github.com/getnamo/tensorflow-ue4/issues/19) that doesn't allow both pip and tensorflow dependencies to be pulled in the same launch, for this reason after pip is installed you need to relaunch the project one extra time. That said you can see it become ready to use on the second launch, in the editor output or python console window, when all the requirements are satisfied. At that time you should see 
![download](https://camo.githubusercontent.com/711214ab75ba5ddabb0629e1b24fa9913680a1a1/687474703a2f2f692e696d6775722e636f6d2f7338574475374d2e706e67) and then you should be good to play in editor. This should obviously be fixed and some GUI progress window for dependency installation would be ideal, but consider this a very early stage plugin for the moment. Contributions are welcome :)

I do believe your path issues may stem from placing directly under the C: drive which may for some reason require admin rights which the pip cmd doesn't have, but I'm not sure. The issue and the easy workaround is definitely a good point to add to the readme.

Accuracy of MNIST recognition is based on the testing and validation sets which use the MNIST database. I believe the dataset uses grayscale drawing (white/offwhite grey on black, but this is correctly inverted for samples sent so that part isn't an issue) vs the all black lines drawn in the example images or the ones sent from the web. That along with slightly different thickness means that it is dissimilar enough to fail detection on simple models. Keras convolutional neural network example on the other hand is much more robust, but I find its accuracy on the data being sent from UE4 in the black only lines is closer to 9/10 which would be 90%, not 98%. A dataset for training with input sets in the exactly same format as those sent for inference or a more varied one with noise and different conditions would ensure that the trained model would categorize entries with higher accuracy and hopefully closer to the accuracy stated by the testing/validation set.

I'll add the video link to the readme in a troubleshooting section, it may help some people having similar issues.

**Comment by getnamo** [comment]
added the troubleshooting section here: https://github.com/getnamo/tensorflow-ue4-examples/blob/master/README.md#troubleshooting

let me know if linking your video there is ok

**Comment by Berranzan** [comment]
It is perfectly OK, thank you for your great work!
